### PR TITLE
[3.12] gh-101100: Fix Sphinx warnings for removed dead batteries (GH-113669)

### PR DESCRIPTION
**Pull request by hugovk**
(cherry picked from commit fab7ad62ceca1f88767bca4e1f06f8e4b1faef2f)

* Issue: gh-101100

----
📚 Documentation preview 📚: https://cpython-previews--113681.org.readthedocs.build/

### CODE REVIEW

File: Doc/whatsnew/3.5.rst [hunk 1]
```diff
@@ -878,7 +878,7 @@ size of decompressed data. (Contributed by Nikolaus Rath in :issue:`15955`.)
 cgi
 ---
 
-The :class:`~cgi.FieldStorage` class now supports the :term:`context manager`
+The :class:`!FieldStorage` class now supports the :term:`context manager`
```
Comment (hugovk) [inline Doc/whatsnew/3.5.rst]:
```suggestion
The :class:`~cgi.FieldStorage` class now supports the :term:`context manager`
```

File: Doc/whatsnew/3.5.rst [hunk 2]
```diff
@@ -1714,7 +1714,7 @@ support :rfc:`6531` (SMTPUTF8).
 sndhdr
 ------
 
-The :func:`~sndhdr.what` and :func:`~sndhdr.whathdr` functions  now return
+The :func:`!what` and :func:`!whathdr` functions  now return
```
Comment (hugovk) [inline Doc/whatsnew/3.5.rst]:
```suggestion
The :func:`~sndhdr.what` and :func:`~sndhdr.whathdr` functions  now return
```

File: Doc/whatsnew/3.7.rst [hunk 3]
```diff
@@ -2304,9 +2304,9 @@ Changes in the Python API
 * The :attr:`struct.Struct.format` type is now :class:`str` instead of
   :class:`bytes`. (Contributed by Victor Stinner in :issue:`21071`.)
 
-* :func:`~cgi.parse_multipart` now accepts the *encoding* and *errors*
+* :func:`!cgi.parse_multipart` now accepts the *encoding* and *errors*
```
Comment (hugovk) [inline Doc/whatsnew/3.7.rst]:
```suggestion
* :func:`~cgi.parse_multipart` now accepts the *encoding* and *errors*
```

**Review by hugovk — COMMENTED** [review]
These can stay in 3.12: https://peps.python.org/pep-0594/#deprecated-modules

**Review by AlexWaygood — APPROVED** [review]
Yup, these are all dead and buried in Python 3.12!

**Pull request merged**